photon-lib/src/main/scala/com/linkedin/photon/ml/evaluation/EvaluatorSet.scala

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.evaluation
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.apache.spark.storage.StorageLevel
+
+import com.linkedin.photon.ml.Types.UniqueSampleId
+import com.linkedin.photon.ml.spark.RDDLike
+
+/**
+ * A [[Set]] of [[Evaluator]] objects and the labels + offsets + weights used to evaluate scored data.
+ *
+ * @param evaluators The set of [[Evaluator]] objects to use to compute validation metrics
+ * @param primaryEvaluator The [[Evaluator]] which computes the most important validation metric (e.g. the metric used
+ *                         for model selection)
+ * @param labelAndOffsetAndWeights A [[RDD]] of (unique ID, (label, offset, weight) tuple), which is joined with a
+ *                                 [[RDD]] of scores to compute validation metrics
+ */
+class EvaluatorSet(
```

Comment:
I was thinking since it contains a `Set` of `Evaluator` objects, but point taken. Let me see how difficult it would be to implement the `Set` interface, otherwise I'll rename the class.

Reply:
@joshvfleming I renamed it to `EvaluationSuite` - it seemed like a better name at the time though I've since become indecisive again.